x/ibc/core/04-channel/keeper/handshake.go

```diff
@@ -96,7 +96,8 @@ func (k Keeper) ChanOpenTry(
 	order types.Order,
 	connectionHops []string,
 	portID,
-	channelID string,
+	channelID,
+	provedChannelID string,
```

Comment:
Personally this is very confusing for me to read. It's not immediately clear what these mean without poring through the spec+comments.

I would prefer `channelID => desiredChannelID` and `provedChannelID => counterpartyChosenChannelID`

cc: @cwgoes

Reply:
agree on the confusion. Why not just `desiredChannelID` and `chosenChannelID`? Using counterparty can be vague to whether the counterparty chose the channel ID for this chain or if the counterparty chose their own. 

Then the logic would be 
```go 
// empty chosen channel id allows the desired channel id to be used for identifier selection
if chosenChannelID != "" && chosenChannelID != desiredChannelID {
return err
}
```

Reply:
I'm fine with this renaming, I can update the spec as well

Reply:
upon further reflection I think `counterpartyChosenChannelID` works fine since in most cases the `Counterparty` will be a different field containing the `counterparty.ChannelId`

Reply:
OK!